glean-core/android/src/test/java/mozilla/telemetry/glean/DispatchersTest.kt

```diff
@@ -60,4 +65,52 @@ class DispatchersTest {
         assertEquals("Tasks have executed", 3, threadCanary)
         assertEquals("Task queue is cleared", 0, Dispatchers.API.taskQueue.size)
     }
+
+    @Test
+    fun `queued tasks are executed in the order they are received`() {
+        val orderedList = mutableListOf<String>()
+        val addTasks = mutableListOf<Job>()
+
+        // Set testing mode to false as we want every call to `launch` to execute as a coroutine
+        Dispatchers.API.setTestingMode(false)
+        Dispatchers.API.setTaskQueueing(true)
+
+        // Spawn a coroutine that will add elements to the orderedList.  This will continue to add
+        // elements to the queue until there are at least 50 elements in the queue. At that point,
+        // the second coroutine below will flush and disable the queue.
+        val addJob = GlobalScope.launch {
+            (0..100).forEach { num ->
+                Dispatchers.API.launch {
+                    orderedList.add("$num")
+                }.also { job ->
```

Comment:
Oh noes, another operator

Reply:
While my original Kotlinism works, I could be more clear.  Let me rephrase it...